src/Microsoft.ML.FastTree/GamTrainer.cs

```diff
@@ -209,25 +96,41 @@ public abstract class ArgumentsBase : LearnerInputBaseWithWeight
         private const string RegisterName = "GamTraining";
 
         //Parameters of Training
-        internal readonly TArgs Args;
+        protected readonly TArgs Args;
         private readonly double _gainConfidenceInSquaredStandardDeviations;
         private readonly double _entropyCoefficient;
 
         //Dataset Information
-        internal Dataset TrainSet;
+        protected Dataset TrainSet;
+        protected Dataset ValidSet;
+        /// <summary>
+        /// Whether a validation set was passed in
+        /// </summary>
+        protected bool HasValidSet => ValidSet != null;
+        protected ScoreTracker TrainSetScore;
+        protected ScoreTracker ValidSetScore;
+        protected TestHistory PruningTest;
+        protected int PruningLossIndex;
         protected int InputLength;
         private LeastSquaresRegressionTreeLearner.LeafSplitCandidates _leafSplitCandidates;
         private SufficientStatsBase[] _histogram;
+        private ILeafSplitStatisticsCalculator _leafSplitHelper;
+        private ObjectiveFunctionBase _objectiveFunction;
+        private bool HasWeights => TrainSet?.SampleWeights != null;
 
-        protected bool HasWeights => TrainSet?.SampleWeights != null;
+        // Training Datastructures
+        private SubGraph _subGraph;
 
         //Results of Training
```

Comment:
>//Results of Training [](start = 8, length = 21)

please use the common syntax for comments.
//<space>Sentence starting from capital letter (no other capitalized words inside)<period>

Reply:
Done for all comments!